src/pytorch_lightning/utilities/argparse.py

```diff
@@ -127,7 +121,7 @@ def parse_env_variables(cls: Type["pl.Trainer"], template: str = "PL_%(cls_name)
     return Namespace(**env_args)
 
 
-def get_init_arguments_and_types(cls: Any) -> List[Tuple[str, Tuple, Any]]:
+def get_init_arguments_and_types(cls: Union[_ARGPARSE_CLS, Any]) -> List[Tuple[str, Tuple, Any]]:
```

Comment:
note to self: remove this Any and check mypy locallly

Reply:
this Any was added to avoid the following error

```sh
src/pytorch_lightning/utilities/argparse.py:222: error: Argument 1 to "get_init_arguments_and_types" has incompatible type "Union[Type[LightningDataModule], Type[Trainer], Callable[[LightningDataModule], None], Callable[[Trainer, Union[Logger, Iterable[Logger], bool], bool, Union[List[Callback], Callback, None], Optional[str], Union[int, float, None], Optional[str], int, Optional[int], Union[List[int], str, int, None], Union[List[int], str, int, None], bool, Union[List[int], str, int, None], Optional[int], bool, Union[int, float], Union[int, float, str], Optional[int], Union[int, bool], Union[int, Dict[int, int], None], Optional[int], Optional[int], int, Optional[int], Union[str, timedelta, Dict[str, int], None], Union[int, float, None], Union[int, float, None], Union[int, float, None], Union[int, float, None], Union[int, float, None], int, Union[str, Accelerator, None], Union[str, Strategy, None], bool, Union[int, str], bool, Optional[str], int, Union[Path, str, None], Union[Profiler, str, None], Optional[bool], Union[bool, Literal['warn'], None], int, Union[bool, str], bool, bool, Union[str, bool], Union[Strategy, PrecisionPlugin, ClusterEnvironment, CheckpointIO, LayerSync, str, List[Union[Union[Strategy, PrecisionPlugin, ClusterEnvironment, CheckpointIO, LayerSync], str]], None], str, Optional[str], bool, str], None]]"; expected "Union[Type[LightningDataModule], Type[Trainer]]"  [arg-type]
```

I ignored that error in favor of removing the Any, in commit https://github.com/Lightning-AI/lightning/pull/13693/commits/cc55ab96c3ac5e7399c29bd77ab8b6ac8217eb2d